### ISSUE TITLE
Ender Dragon and Return Portal Don't Spawn in Brand-New World if Setting "scan-for-legacy-ender-dragon" is Set To False in paper.yml

### ISSUE THREAD

**Issue by IAmEchino** [PaperSpigot/Paper#6991]
### Expected behavior

I expect the ender dragon and the (unlit) return portal to spawn upon entering the end dimension. This was observed on multiple server members' computers **in Singleplayer mode.**

![2021-12-02_01 58 05](https://user-images.githubusercontent.com/64720141/144373337-49aaa972-498c-46c2-9c62-207ebe008f16.png)

### Observed/Actual behavior

The dragon did not spawn and neither did the return portal.

This has been attempted multiple times. Results were identical for both 10 chunk and 12 chunk simulation distances.

![2021-12-02_01 50 42](https://user-images.githubusercontent.com/64720141/144372364-87a81d3c-b791-464e-9d6c-add80cb0d36e.png)

### Steps/models to reproduce

Set `scan-for-legacy-ender-dragon` to `false` in paper.yml.

~~In server configuration, set simulation distance to 10 or 12 (I tested both) and set render distance to 18~~

Enter the end through any portal (constructed or naturally generated)

The dragon does not spawn and neither does the return portal.

### Plugin List

BetterConcrete

CoreProtect

EssentialsX

EssentialsChat

EssentialsProtect

LuckPerms

RHLeafDecay

### Datapack List

Vanilla (built-in)

file/bukkit (world)

### Paper version

[01:33:26 INFO]: Checking version, please wait...
[01:33:26 INFO]: This server is running Paper version git-Paper-21 (MC: 1.18) (Implementing API version 1.18-R0.1-SNAPSHOT) (Git: a5a7782)
You are running the latest version

### Other

EssentialsX and all related plugins are up to date. EssentialsProtect's option to prevent the ender dragon from spawning is disabled.

**Comment by henrikx** [comment]
![image](https://user-images.githubusercontent.com/10342989/144419406-d4d228e6-5e90-4c53-9e60-3740a3e0f545.png)
Could not reproduce.

Edit:
Simulation distance 10 for this test. View distance was 10. Will test again with 18 view-distance.
Edit 2:
Still the same result. Could not reproduce.

**Comment by e-im** [comment]
With what seed did you observe this behaviour? As it sounds like you could consistently reproduce it, does this happen without plugins on the server?

**Comment by IAmEchino** [comment]
The seed was -5859561077280550856.

I just tested it without any plugins with the same result.

I also tested without any plugins with the seed "PaperMC" and got the same result.

**Comment by IAmEchino** [comment]
### I FIXED IT!

I removed each of my config files one by one until the problem was fixed. It ended up being my paper.yml. 

I'd appreciate it if y'all had a look at this thing to see if there's anything wrong with it. If there is, I'd like to know what, but if not, maybe it's a bug?

Here it is, pasted in full: https://pastebin.com/SkrPEbq9

**Comment by ghost** [comment]
@CommandoEchino 
scan-for-legacy-ender-dragon: false
Put that back to true, that should fix it

**Comment by IAmEchino** [comment]
@Clashbestie Yes, that worked, thank you..._but why?_ This was a newly generated world! Is this a bug or a feature?

**Comment by ghost** [comment]
@CommandoEchino I think its a bug, if not they should change the name.

**Comment by IAmEchino** [comment]
I tested a world with a near-default paper.yml with the only difference being that `scan-for-legacy-ender-dragon` is set to `false` in paper.yml. The dragon and return portal did not spawn. Therefore this has to be the issue, and I've updated the post and the title to reflect as such.

**Comment by e-im** [comment]
So, this is intentional behaviour; with this set to false, no attempt will be made to set up the fight. However, I don't believe having this option is necessary anymore, or at least the concerns expressed in #941 are no longer valid.

Unless anyone else has any reason to use this feature, I don't see it being necessary.

**Comment by IAmEchino** [comment]
I understand your opinion that this is intentional behavior, but in my opinion, deciding to not check for a legacy fight should certainly not break the functionality of a brand-new world, and makes this tantamount to a bug.

I absolute agree with @Clashbestie that if the actual behavior of this is to remain the same, the _name_ of the option should be changed to `set-up-ender-dragon-fight` or something like that to avoid confusion.
